docs/lib/datastore/fragments/flutter/getting-started/50_initDataStore.md

```diff
@@ -16,28 +14,38 @@ Your code should look like this:
 ```dart
 import 'package:amplify_flutter/amplify.dart';
 import 'package:amplify_datastore/amplify_datastore.dart';
-import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:flutter/widgets.dart';
 
 import 'amplifyconfiguration.dart';
 import 'models/ModelProvider.dart';
 
 class MyAmplifyApp extends StatefulWidget {
+  @override
+  _MyAmplifyAppState createState() => _MyAmplifyAppState();
+}
 
-    @override
-    void initState() {
-        super.initState();
-        _configureAmplify();
+class _MyAmplifyAppState extends State<MyAmplifyApp> {
+  @override
+  void initState() {
+    super.initState();
+    _configureAmplify();
+  }
+
+  void _configureAmplify() async {
+    // Add the following line to add DataStore plugin to your app
+    Amplify.addPlugin(AmplifyDataStore(modelProvider: ModelProvider.instance));
+
+    try {
+      await Amplify.configure(amplifyconfig);
+    } on AmplifyAlreadyConfiguredException {
+      print(
+          "Tried to reconfigure Amplify; this can occur when your app restarts on Android.");
     }
```

Comment:
Didn't we move away from this to `isConfigured`? cc: @haverchuck

Reply:
Do you mean setting an `isConfigured` value and then removing the try catch?

This is how most of the examples are in the docs, but if there is another approach that is preferable I am happy to update them all. It actually would probably be good to decide on a consistent example of how to configure amplify. In searching around the docs for other examples I am realizing this is not the only one that doesn't correctly create a StatefulWidget.

The one example I found that has a boolean value for configured is [this](https://github.com/aws-amplify/docs/blob/master/docs/start/getting-started/fragments/flutter/integrate.md), but it still has the try catch.

Reply:
@Amplifiyer I tested this out with the hot restart changes. `Amplify.isConfigured` will be false after a hot restart, so the customer will still need to handle the exception for hot restarts.

I think `Amplify.isConfigured` being false after a hot restart is expected since that state will be lost. It *might* be ideal to have it maintain its state, but I don't think that was intended to be part of the hot restart fix. @haverchuck correct me if I am mistaken.

I think we should leave the try/catch, but update the print statement since I think most users will see this on hot restart now that hot restart works. I don't know the history of the current message ("this can occur when your app restarts on Android"). I assume that means re-launching after being moved to the background, but I am not sure.

My suggestion would be to update it to the following, but am open to suggestions. I took the Android portion out. If it is still relevant I can add it back, but I think we should clarify what "restarts on Android" means. As someone that is new to amplify, that wasn't clear to me. Once we decide, I can update all the docs to be consistent.

```dart
try {
  await Amplify.configure(amplifyconfig);
} on AmplifyAlreadyConfiguredException {
  print("Amplify was already configured. This can occur after a hot reload.");
}
```

Reply:
We continue to throw (and not swallow) the AmplifyAlreadyConfiguredException because attempting to call configure multiple times is something customers might actually try (I think this is supported in JS?)... but now that I am revisiting the code, I think we could swallow the exception on the method channel response since redundant calls would only make it across the method channel on a hot restart.  But in the absense of making that change (which would admittedly be small) then the change proposed above makes sense.

Reply:
I decided to make this PR more targeted to the original issue I was trying to fix (incorrect import & inconsistent configuration) since we should probably update all the docs after [this PR](https://github.com/aws-amplify/amplify-flutter/pull/527) lands.